qiskit/transpiler/preset_passmanagers/__init__.py

```diff
@@ -20,13 +20,180 @@
+        initial_layout (Layout): Initial position of virtual qubits on
+            physical qubits.
+        layout_method (str): the :class:`~.Pass` to use for choosing initial qubit
+            placement. Valid choices are ``'trivial'``, ``'dense'``, ``'noise_adaptive'``,
+            and, ``'sabre'`` repsenting :class:`~.TrivialLayout`, :class:`~DenseLayout`,
+            :class:`~.NoiseAdaptiveLayout`, :class:`~.SabreLayout` respectively.
+        routing_method (str): the pass to use for routing qubits on the
+            architecture. Valid choices are ``'basic'``, ``'lookahead'``, ``'stochastic'``,
+            ``'sabre'``, and ``'none'`` representing :class:`~.BasicSwap`,
+            :class:`~.LookaheadSwap`, :class:`~.StochasticSwap`, :class:`~.SabreSwap`, and
+            erroring if routing is required respectively.
+        translation_method (str): the method to use for translating gates to
+            basis_gates. Valid choices ``'unroller'``, ``'translator'``, ``'synthesis'``
+            representing :class:`~.Unroller`, :class:`~.BasisTranslator`, and
+            :class:`~.UnitarySynthesis` respectively.
+        scheduling_method (str): the pass to use for scheduling instructions. Valid choices
+            are ``'alap'`` and ``'asap``.
+        backend_properties (BackendProperties): Properties returned by a
+            backend, including information on gate errors, readout errors,
+            qubit coherence times, etc.
+        approximation_degree (float): heuristic dial used for circuit approximation
+            (1.0=no approximation, 0.0=maximal approximation)
+        seed_transpiler (int): Sets random seed for the stochastic parts of
+            the transpiler.
+        unitary_synthesis_method (str): The name of the unitary synthesis
+            method to use. By default 'default' is used, which is the only
+            method included with qiskit. If you have installed any unitary
+            synthesis plugins you can use the name exported by the plugin.
+        unitary_synthesis_plugin_config (dict): An optional configuration dictionary
+            that will be passed directly to the unitary synthesis plugin. By
+            default this setting will have no effect as the default unitary
+            synthesis method does not take custom configuration. This should
+            only be necessary when a unitary synthesis plugin is specified with
+            the ``unitary_synthesis`` argument. As this is custom for each
+            unitary synthesis plugin refer to the plugin documentation for how
+            to use this option.
+
+    Returns:
+        PassManager: The preset pass manager for the given options
+
+    Raises:
+        ValueError: if an invalid value for ``optimization_level`` is passed in.
+    """
+
+    if backend is not None:
+        pm_config = PassManagerConfig.from_backend(
+            backend,
+            target=target,
```

Comment:
This is a suggestion you're very welcome to ignore, but would this be a bit less prone to forgetting an argument in either of the if-clauses if we'd write it something like the following?
```python
settings = {"target": target, "basis_gates": basis_gates, ...}
if backend is not None:
    pm_config = PassManagerConfig.from_backend(backend, **settings)
else:
    pm_config = PassManagerConfig(**settings)
```

Reply:
Good call, done in: https://github.com/Qiskit/qiskit-terra/pull/8125/commits/4e86d95a2324e36932b48bb12206d421a511542e